src/FrameId.h

```diff
@@ -1,10 +1,37 @@
 #ifndef FRAMEID_H_INCLUDED
 #define FRAMEID_H_INCLUDED
 
-typedef int FrameId;
+#include <cstddef>
+
+struct FrameId {
+	static constexpr int Invalid = -1;
+	constexpr FrameId() :
+		m_id(Invalid) {}
+	constexpr FrameId(int new_id) :
+		m_id(new_id) {}
+
+	constexpr operator bool() const { return m_id > Invalid; }
+	constexpr operator int() const { return m_id; }
+	constexpr operator size_t() const { return m_id; }
+
+	constexpr bool operator==(FrameId rhs) const { return m_id == rhs.m_id; }
+	constexpr bool operator!=(FrameId rhs) const { return m_id != rhs.m_id; }
+
+	constexpr bool valid() const { return m_id > Invalid; }
+	constexpr int id() const { return m_id; }
+
+private:
+	int m_id;
+};
+
+static_assert(sizeof(FrameId) == sizeof(int) && alignof(FrameId) == alignof(int),
+	"Error: FrameId sized differently than the underlying type on this platform!");
 
 constexpr FrameId noFrameId = -1;
```

Comment:
would it be better to move `noFrameId` definition on top and use it to define `FrameId.Invalid`?

As:
```

constexpr FrameId noFrameId = -1;

struct FrameId {
	static constexpr int Invalid = noFrameId;
```

Reply:
Somewhere between writing and bugfixing, I forgot that I intended to go through the rest of the code and get rid of IsIdValid and noFrameId. The current incarnation is just a patch to keep the rest of the code working while I rewrote FrameId.